File: frontend/web-proje/src/services/productService.js
import axios from 'axios'

export default class ProductService {

    getAll(){
        return axios.get(`http://localhost:8080/products/getAll`);
    }

    getByNameConstains(name){
        return axios.get(`http://localhost:8080/products/getByNameConstains?name=${name}`)
    }

    getById(id){
        return axios.get(`http://localhost:8080/products/getById?id=${id}`);
    }

    getByCustomerId(customerId){
        return axios.get(
            `http://localhost:8080/products/getByCustomer_Id?customerId=${customerId}`
            );
    }

    add(product){
        axios.post(
            "http://localhost:8080/products/add",
            product
            )
    }

}

File: backend/webproje/src/main/java/webprogramlama/demo/service/abstracts/ProductService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package webprogramlama.demo.service.abstracts;

import java.util.List;
import webprogramlama.demo.entity.product.Product;

/**
 *
 * @author kaan
 */

public interface ProductService {
    
    List<Product> getAll();
    
    List<Product> getByCustomer_Id(int customerId);

    Product getById(int productId);

    List<Product> getByProductNameContains(String productName);
    
    void insert(Product product);
    
    void delete(Product product);
}
